ui/src/api/config.service.ts

```diff
@@ -0,0 +1,24 @@
+import {ClientConfig} from './client-config';
+import {API_BY_PAGE_HOST} from '../constants';
+
+export class ConfigService {
+  private readonly apiUrl: string;
+
+  constructor() {
+    this.apiUrl = API_BY_PAGE_HOST[window.location.host];
+  }
+
+  async getConfig(): Promise<ClientConfig | null> {
+    const headers = {};
+    try {
+      const response = await fetch(`${this.apiUrl}/client-config`, {
+        method: 'GET',
+        headers: headers,
```

Comment:
Are we expecting any headers in the future? If not, this field & the `headers` variable could just be removed.